multitool/private/multitool.bzl

```diff
@@ -60,14 +60,15 @@ def _check(condition, message):
     if not condition:
         fail(message)
 
-def _check_version(os):
+def _check_version(os, binary_os):
+    # require bazel 7.1 on windows. Only do this check for windows artifacts to avoid regressing anyone
     # skip version check on windows if we don't have a release version. We can't tell from a hash what features we have.
-    if os == "windows" and native.bazel_version:
+    if os == "windows" and binary_os == "windows" and native.bazel_version:
         version = native.bazel_version.split(".")
         if int(version[0]) > 7 or (int(version[0]) == 7 and int(version[1]) >= 1):
```

Comment:
Note: the versions.bzl utility in Bazel-skylib might be appropriate here (if the ruleset already depends on it)